### ISSUE TITLE
RFLCT_Ambiguous in release mode

### ISSUE THREAD

**Issue by SamuelLioult** [dotnet/efcore#6242]
Hi,

I develop a UWP application and I get a System.Reflection.AmbiguousMatchException when I try to add an element to the db. My class hasn't any linked class (no foreign key).
I use the Add method inherited from DbContext.

I get this issue only in Release mode, but not in Debug mode.

```
Exception message: "RFLCT_Ambiguous"
Stack trace:
at System.SharedTypeExtensions.GetAnyProperty(Type type, String name)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.SnapshotFactoryFactory.CreateSnapshotExpression(Type entityType, ParameterExpression parameter, Type[] types, IList`1 propertyBases)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.SnapshotFactoryFactory.CreateConstructorExpression(IEntityType entityType, ParameterExpression parameter)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.SnapshotFactoryFactory`1.Create(IEntityType entityType)
at Microsoft.EntityFrameworkCore.Metadata.Internal.EntityType.<>c.<get_OriginalValuesFactory>b__141_0(EntityType entityType)
at Microsoft.EntityFrameworkCore.Metadata.Internal.EntityType.get_OriginalValuesFactory()
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.OriginalValues..ctor(InternalEntityEntry entry)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.EnsureOriginalValues()
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntrySubscriber.SnapshotAndSubscribe(InternalEntityEntry entry)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.StartTracking(InternalEntityEntry entry)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState oldState, EntityState newState, Boolean acceptChanges)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState entityState, Boolean acceptChanges)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.PaintAction(EntityEntryGraphNode node)
at System.Predicate`1.InvokeOpenStaticThunk(T obj)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph(EntityEntryGraphNode node, Func`2 handleNode)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.AttachGraph(InternalEntityEntry rootEntry, EntityState entityState)
at Microsoft.EntityFrameworkCore.DbContext.SetEntityState(InternalEntityEntry entry, EntityState entityState)
at Microsoft.EntityFrameworkCore.DbContext.SetEntityState[TEntity](TEntity entity, EntityState entityState)
at Microsoft.EntityFrameworkCore.DbContext.Add[TEntity](TEntity entity)
```
### Further technical details

EF Core version: v 1.0.0
Operating system: Windows 10 Mobile v 10.0.14393.5
Visual Studio version: VS Community 2015

Thanks

**Comment by SamuelLioult** [comment]
According to this post http://stackoverflow.com/questions/18245686/entity-framework-exception-ambiguous-match-found
I done a workaround to pass the problem. (But I don't understand why it works fine in Debug mode...)

Now, I get a new issue, in Release mode:

> The 'Id' on entity type 'MyEntity' does not have a value set and no value generator is available for properties of type 'Int32'. Either set a value for set the property before adding the entity or configure a value generator for properties of type 'Int32'.

Here is the stack trace:

`at Microsoft.EntityFrameworkCore.ValueGeneration.ValueGeneratorSelector.Create(IProperty property, IEntityType entityType)
at Microsoft.EntityFrameworkCore.ValueGeneration.RelationalValueGeneratorSelector.Create(IProperty property, IEntityType entityType)
at Microsoft.EntityFrameworkCore.ValueGeneration.ValueGeneratorCache.<>c.<GetOrAdd>b__2_0(CacheKey ck)
at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
at Microsoft.EntityFrameworkCore.ValueGeneration.ValueGeneratorCache.GetOrAdd(IProperty property, IEntityType entityType, Func`3 factory)
at Microsoft.EntityFrameworkCore.ValueGeneration.ValueGeneratorSelector.Select(IProperty property, IEntityType entityType)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.ValueGenerationManager.Generate(InternalEntityEntry entry)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.InternalEntityEntry.SetEntityState(EntityState entityState, Boolean acceptChanges)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.PaintAction(EntityEntryGraphNode node)
at System.Collections.Generic.TreeWalkPredicate`1.InvokeOpenStaticThunk(Node node)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityEntryGraphIterator.TraverseGraph(EntityEntryGraphNode node, Func`2 handleNode)
at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.EntityGraphAttacher.AttachGraph(InternalEntityEntry rootEntry, EntityState entityState)
at Microsoft.EntityFrameworkCore.DbContext.SetEntityState(InternalEntityEntry entry, EntityState entityState)
at Microsoft.EntityFrameworkCore.DbContext.SetEntityState[TEntity](TEntity entity, EntityState entityState)
at Microsoft.EntityFrameworkCore.DbContext.Add[TEntity](TEntity entity)`

**Comment by divega** [comment]
@natemcmaster possible bug in .NET Native.

**Comment by natemcmaster** [comment]
@SamuelLioult your issue doesn't contain enough information for us to reproduce this problem and it's not a known issue I recognize. Can you you share more information?

```
### Steps to reproduce
Ideally include a complete code listing that we can run to reproduce the issue.
Alternatively, you can provide a project/solution that we can run.
```

A [zipped-up](https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/) project is best as this is most definitely a .NET Native issue. I'll need to know how the .NET Native compiler is behaving to understand reasons for the error.

Also, the stackoverflow post you referenced is probably a red herring: EF 5 and EF Core have a very different stack.

**Comment by natemcmaster** [comment]
Closing as we haven't heard back yet on the request for more details. We can reopen for more investigation if we can get a repro project.

Btw, I'm not closing just to be mean. We got a lot of traffic and bug reports. It's hard to for us reproduce errors with only fragments of information. To maximize our time building features and investigating bugs, we ask for folks to give us a self contained way to reproduce the issue.
